### PR TITLE
Add note to call attention to the changed instructions

### PR DESCRIPTION
**Pull request by umanwizard**
Discussed on Slack: https://materializeinc.slack.com/archives/C015RHB3LDR/p1623790505116300?thread_ts=1623790219.114700&cid=C015RHB3LDR

### CODE REVIEW

File: doc/user/content/install.md [hunk 1]
```diff
@@ -54,6 +54,9 @@ curl -L https://downloads.mtrlz.dev/materialized-{{< version >}}-x86_64-apple-da
 
 Run the following commands as root.
 
+**Note!** These instructions changed between versions 0.8.0 and 0.8.1. If you ran them
+previously, you may need to do so again to continue receiving updates.
```
Comment (quodlibetor) [inline doc/user/content/install.md:58]: hmm this probably should be in a `note` block but that doesn't exist, and it probably doesn't deserve to be in a [`{{ <warning> }}` block](https://github.com/MaterializeInc/materialize/blob/main/doc/user/layouts/shortcodes/warning.html), so I guess this LGTM.

Comment (umanwizard) [reply doc/user/content/install.md:58]: I'll wait and see if Sean or Micole has input on how best to format this before landing

**Review by quodlibetor — COMMENTED** [review]
I think we generally try to have semantic markup for this kind of thing, but since we don't have anything with the correct semantics I'm inclined to not block this on adding an appropriate shortcode and design for it. OTOH this isn't urgent until we do another release or bintray shuts down.

**Comment by umanwizard** [comment]
@quodlibetor I actually want to land it ASAP since I've already linked to this doc in the Community slack

File: doc/user/content/install.md (hunk 1 above)
Comment (sploiselle) [reply doc/user/content/install.md:58]: What is the error folks will run into if they use the "old" command in the new environment?

Comment (umanwizard) [reply doc/user/content/install.md:58]: Today they will get v0.8.0 (and never get any new updates); if they run this past July 4th (when our old host is being turned off) they will get a 404 error or similar.

The biggest issue is that people who have already run this in the past will start getting errors too, then navigate to our docs to try to figure out what's going on, and we need to make sure they see this and know they need to run the new instructions.

**Comment by umanwizard** [comment]
Landing for now since I already mentioned this on the Community slack and want to get _some_ clarifying text out there; we can "fix it in post" if someone has a better strategy for formatting!

**Pull request merged**